compiler/luci/service/src/CircleShapeSignatureInferenceHelper.cpp

```diff
@@ -16,12 +16,130 @@
 
 #include "luci/Service/CircleShapeSignatureInferenceHelper.h"
 
+#include <loco.h>
+
+#include <luci/Log.h>
+
+#include <oops/InternalExn.h>
+
+namespace
+{
+
+luci::ShapeSignature clean_signature(const luci::ShapeSignature &signature)
+{
+  // If shape signature has at least one -1, it is not static.
+  for (uint32_t i = 0; i < signature.rank(); ++i)
+    if (signature.dim(i) == -1)
+      return signature;
+
+  // If all dimensions are static, return empty shape signature.
+  return luci::ShapeSignature();
+}
+
+} // namespace
+
 namespace luci
 {
 
 namespace ssinf
 {
 
+ShapeSignature reduced_signature(const loco::Node *node, const loco::Node *indices, bool keep_dims)
```

Comment:
(this is about the naming convention) below used `signature_of_arg` and this differs from `X_of_Y` type
I propose to revise `signature_of_arg` to `arg_signature` to follow `reduced_signature`
- as `arg_signature` seems somewhat strange, I think `input_arg_signature` would be better to the method to express itself.

What do you think?

Reply:
I think `input_arg_signature` is good idea! Naming is always difficult for me :(

Thanks for your good idea and I will create another PR for applying it :-)